lms/envs/common.py

```diff
@@ -1219,6 +1219,7 @@
     'js/vendor/jquery.cookie.js',
     'js/vendor/url.min.js',
     'js/vendor/underscore-min.js',
+    'js/vendor/underscore.string.min.js',
```

Comment:
Note to reviewers: I decided to move this into base_vendor_js so that we could be guaranteed that it loads BEFORE requirejs. Previously it sometime loaded independently of requiresjs (meaning that it was defined as _.str), and other times was loaded by requirejs (meaning that it was not stored in the global namespace). This caused a lot of confusing code like this "str = str || _.str".